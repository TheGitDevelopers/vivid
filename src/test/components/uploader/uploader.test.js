import React from 'react';
import renderer from 'react-test-renderer';
import Uploader from '../../../components/common/uploader/uploader';


describe('Uploader tests:', () => {
  /* Test mockups */
  const onChangeMock = jest.fn();
  const uploadTextMock = 'content';
  const tree = renderer
    .create(<Uploader uploadText={uploadTextMock} onChange={onChangeMock} />);

  /* All component tests should be listed here */
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('has prop named "uploadText"', () => {
    expect(tree.root.findByType(Uploader).props.uploadText).toBe(uploadTextMock);
  });

  it('has prop function named "onChange" ', () => {
    tree.root.findByType(Uploader).props.onChange();
    expect(onChangeMock.mock.calls.length).toBe(1);
  });
});
