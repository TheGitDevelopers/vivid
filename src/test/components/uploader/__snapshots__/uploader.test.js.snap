// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader tests: renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F0F4F8",
      "borderRadius": 8,
      "elevation": 2,
      "height": 670,
      "justifyContent": "center",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.32,
      "shadowRadius": 5.46,
      "width": 670,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#A6BCD0",
        "fontFamily": "lato-regular",
        "fontSize": 16,
      }
    }
  >
    content
  </Text>
</View>
`;
